mlflow/tensorflow/__init__.py

```diff
@@ -887,6 +968,110 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
+                    input_training_data_original = args[0]
+                    input_example_slice = None
+
+                    def _extract_from_sequence_or_generator():
+                        input_training_data = input_training_data_original
+                        if isinstance(input_training_data_original, types.GeneratorType):
+                            input_training_data = next(input_training_data_original)[0]
+                        elif isinstance(
+                            input_training_data_original, tensorflow.keras.utils.Sequence
+                        ):
+                            input_training_data = input_training_data_original[:][0]
+                        return input_training_data
+
+                    input_training_data = _extract_from_sequence_or_generator()
+                    if isinstance(input_training_data, np.ndarray):
+                        input_example_slice = input_training_data[:INPUT_EXAMPLE_SAMPLE_ROWS]
+                    elif isinstance(input_training_data, tensorflow.data.Dataset):
+                        steps = 1
+                        if history.params is not None and "steps" in history.params:
+                            steps = history.params["steps"]
+
+                        def _extract_n_steps(input_example_n_steps):
+                            if steps > 1:
+                                return np.array(
+                                    [
+                                        v[0][:INPUT_EXAMPLE_SAMPLE_ROWS]
+                                        for v in input_example_n_steps.take(1).as_numpy_iterator()
+                                    ]
+                                )[0]
+                            return np.array(
+                                [
+                                    v[0]
+                                    for v in input_example_n_steps.take(
+                                        INPUT_EXAMPLE_SAMPLE_ROWS
+                                    ).as_numpy_iterator()
+                                ]
+                            )
+
+                        input_example_slice = _extract_n_steps(input_training_data)
+                    elif isinstance(input_training_data, dict):
+                        input_example_slice = {
+                            k: np.take(v, range(0, INPUT_EXAMPLE_SAMPLE_ROWS))
+                            for k, v in input_training_data.items()
+                        }
+                    else:
+                        warnings.warn(
+                            "Tensorflow keras autologging only "
+                            "supports input types of: numpy.ndarray, "
+                            "dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
+                            "or tensorflow.keras.utils.Sequence"
+                        )
+
+                    return input_example_slice
+
+                def _infer_model_signature(input_data_slice):
+                    try:
```

Comment:
`resolve_input_example_and_signature` already does error handling for signature inference failures (see https://github.com/mlflow/mlflow/blob/01182994ec31966dbe8528603d9e1a0ed2188234/mlflow/utils/autologging_utils/__init__.py#L168-L175), so we don't need to try / catch in this method.

Reply:
done. thx